WEB-INF/classes/gov/noaa/pfel/erddap/jte/TableOptions.java

```diff
@@ -0,0 +1,225 @@
+
+    public TableOptionsBuilder(Table table) {
+      this.table = table;
+    }
+
+    public TableOptionsBuilder otherClasses(String otherClasses) {
+      this.otherClasses = otherClasses;
+      return this;
+    }
+
+    public TableOptionsBuilder bgColor(String bgColor) {
+      this.bgColor = bgColor;
+      return this;
+    }
+
+    public TableOptionsBuilder border(int border) {
+      this.border = border;
+      return this;
+    }
+
+    public TableOptionsBuilder writeUnits(boolean writeUnits) {
+      this.writeUnits = writeUnits;
+      return this;
+    }
+
+    public TableOptionsBuilder timeColumn(int timeColumn) {
+      this.timeColumn = timeColumn;
+      return this;
+    }
+
+    public TableOptionsBuilder needEncodingAsHtml(boolean needEncodingAsHtml) {
+      this.needEncodingAsHtml = needEncodingAsHtml;
+      return this;
+    }
+
+    public TableOptionsBuilder allowWrap(boolean allowWrap) {
+      this.allowWrap = allowWrap;
+      return this;
+    }
+
+    public TableOptions build() {
+      return new TableOptions(this);
+    }
+  }
+
+  public String getNoData() {
+    return MustBe.THERE_IS_NO_DATA;
+  }
+
+  public String getWrap() {
+    return allowWrap ? "" : " nowrap";
+  }
+
+  public String getOtherClasses() {
+    return (otherClasses == null || otherClasses.length() == 0 ? "" : " " + otherClasses);
+  }
+
+  public String getBgColor() {
+    return bgColor == null ? "false" : bgColor;
+  }
+
+  public int getNColumns() {
+    return table.nColumns();
+  }
+
+  public int getNRows() {
+    return table.nRows();
+  }
+
+  public String insideForLoophead(int col) {
+    String s = table.getColumnName(col);
+    if (needEncodingAsHtml) {
+      s = XML.encodeAsHTML(s);
+    }
+    if (!somethingWritten && s.trim().length() > 0) {
+      somethingWritten = true;
+    } else if (col == getNColumns() - 1 && s.trim().length() == 0) {
+      s = "&nbsp;";
+    }
+
+    Attributes catts = table.columnAttributes(col);
+    fileAccessBaseUrl[col] = catts.getString("fileAccessBaseUrl");
+    fileAccessSuffix[col] = catts.getString("fileAccessSuffix");
+    if (!String2.isSomething(fileAccessBaseUrl[col])) fileAccessBaseUrl[col] = "";
+    if (!String2.isSomething(fileAccessSuffix[col])) fileAccessSuffix[col] = "";
+
+    return s;
+  }
+
+  public boolean getWriteUnits() {
+    return writeUnits;
+  }
+
+  public String insideForLoopUnits(int col) {
+    String tUnits = table.columnAttributes(col).getString("units");
+    if (col == timeColumn) tUnits = "UTC";
+    if (tUnits == null) tUnits = "";
+    if (needEncodingAsHtml) tUnits = XML.encodeAsHTML(tUnits);
+    if (!somethingWritten && tUnits.trim().length() > 0) {
+      somethingWritten = true;
+    } else if (col == getNColumns() - 1 && tUnits.trim().length() == 0) {
+      tUnits = "&nbsp;";
+    }
+    return tUnits;
+  }
+
+  public int getTimeColumn() {
+    return timeColumn;
+  }
+
+  public String getDoubleDataConverted(int col, int row) {
```

Comment:
```suggestion
  public String getTimeString(int col, int row) {
```

Method rename for clarity